compiler/basicTypes/Multiplicity.hs

```diff
@@ -197,3 +201,29 @@ submultMaybe r1 r2 = go r1 r2
     go One   _     = Smaller
 --    go (MultThing t) (MultThing t') = Unknown
     go _     _     = Unknown
+
+
+traverseMult :: (Multable t, Multable u, Applicative f) => (t -> f u) -> GMult t -> f (GMult u)
+traverseMult _ Zero = pure Zero
+traverseMult _ One = pure One
+traverseMult _ Omega = pure Omega
+traverseMult f (MultThing t) = MultThing <$> f t
+traverseMult f (MultAdd x y) = MultAdd <$> traverseMult f x <*> traverseMult f y
+traverseMult f (MultMul x y) = MultMul <$> traverseMult f x <*> traverseMult f y
+
+multThingList :: Multable t => (t -> a) -> GMult t -> [a]
+multThingList _ Zero = []
+multThingList _ One = []
+multThingList _ Omega = []
+multThingList f (MultThing t) = [f t]
+multThingList f (MultAdd x y) = multThingList f x ++ multThingList f y
```

Comment:
It probably doesn't matter because of how small our multiplicty expressions will be in practice, but I still flinch when I see a recursive call to the left of a `++`.